tests/EdgeToEdge/Routes/SofortPaymentNotificationRouteTest.php

```diff
@@ -119,6 +120,26 @@ public function testGivenValidRequest_applicationIndicatesSuccess(): void {
 		} );
 	}
 
+	public function testGivenValidRequest_donationStateIsChangedToBooked(): void {
+		$this->newEnvironment( function ( Client $client, FunFunFactory $factory ): void {
+			$donation = ValidDonation::newIncompleteSofortDonation();
+			$factory->getDonationRepository()->storeDonation( $donation );
+
+			$client->request(
+				Request::METHOD_POST,
+				'/sofort-payment-notification?id=' . $donation->getId() . '&updateToken=' . self::VALID_TOKEN,
+				[],
+				[],
+				[],
+				$this->buildRawRequestBody( self::VALID_TRANSACTION_ID, self::VALID_TRANSACTION_TIME )
+			);
+
+			$donation = $factory->getDonationRepository()->getDonationById( $donation->getId() );
```

Comment:
*shivers*

```php
$this->assertSame(
    Donation::STATUS_EXTERNAL_BOOKED
    $factory->getDonationRepository()->getDonationById( $donation->getId() )->getStatus()
);
```

No need to have local state. In this case it reuses an existing variable with is very ick